content/en/tutorials/using-terraform-with-testcontainers-and-localstack/index.md

```diff
@@ -0,0 +1,372 @@
+
+
+#### Docker compose
+
+Another way of starting LocalStack with the desired services is using `docker compose`. In the root folder, you'll find the essential configs in the `docker-compose.yml` file:
+<details>
+<summary><b>Expand file</b></summary>
+{{< command >}}
+version: "3.8"
+
+services:
+    localstack:
+        container_name: "${LOCALSTACK_DOCKER_NAME:-localstack-main}"
+        image: localstack/localstack-pro:latest  # required for Pro
+        ports:
+            - "127.0.0.1:4566:4566"            # LocalStack Gateway
+            - "127.0.0.1:4510-4559:4510-4559"  # external services port range
+            - "127.0.0.1:443:443"              # LocalStack HTTPS Gateway (Pro)
+        environment:
+            # Activate LocalStack Pro: https://docs.localstack.cloud/getting-started/auth-token/
+            - LOCALSTACK_AUTH_TOKEN=${LOCALSTACK_AUTH_TOKEN:?}  # required for Pro
+            # LocalStack configuration: https://docs.localstack.cloud/references/configuration/
+            - DEBUG=1
+            - PERSISTENCE=${PERSISTENCE:-0}
+            - EXTENSION_AUTO_INSTALL=localstack-extension-terraform-init
+        volumes:
+            - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+            - "/var/run/docker.sock:/var/run/docker.sock"
+            - "./terraform:/etc/localstack/init/ready.d"
+            - "./target/product-lambda.jar:/etc/localstack/init/ready.d/target/product-lambda.jar"
+{{< / command >}}
+
+</details>
+
+Environment Variables:
+- **LOCALSTACK_AUTH_TOKEN**: Required for using LocalStack Pro.
+- **DEBUG**: Set to 1 to enable verbose logging of the container.
+- **EXTENSION_AUTO_INSTALL**: Automatically installs specified LocalStack [extensions](https://docs.localstack.cloud/user-guide/extensions/), in this case, `localstack-extension-terraform-init` which allows Terraform files to be directly used as init hooks.
+
+Volumes:
+- Docker Socket: Mounts the Docker socket `/var/run/docker.sock` from the host into the container. This allows LocalStack to manage Docker containers directly, facilitating functionalities like spinning up Lambda containers.
+- Terraform Configuration: Mounts a directory containing Terraform files (./terraform) from the host to `/etc/localstack/init/ready.d` in the container. This enables the use of init hooks, as well as the AWS provider (plugins and modules) which is downloaded once and reused in subsequent startups.
+- Lambda Function JAR: Places the `product-lambda.jar` file from the host into the `/etc/localstack/init/ready.d/target` directory in the container, making it available for use, as described in `main.tf`.
+
+After running `docker compose up`, we should keep an eye on the container logs until the `Ready.` message appears.
+
+Now we can test the functionality of our stack by running the following commands:
+
+```bash
+$ aws apigateway get-rest-apis --query 'items[?name==`product-api-gateway`].id' --endpoint http://localhost.localstack.cloud:4566
```

Comment:
Why not use `awslocal`?

Reply:
so the list of things to install is not too long